docs/design.md

```diff
@@ -82,59 +86,115 @@ library or framework that they instrument. Some examples include:
+
+- [**CLR Profiler DLL**](../src/OpenTelemetry.AutoInstrumentation.Native):
+Native component that implements a CLR Profiler. The CLR Profiler is used to
+modify the application [intermediate language](https://en.wikipedia.org/wiki/Common_Intermediate_Language)
+(IL), including the IL of packages used by the application, to add and collect
+observability data.
+On the .NET Framework the CLR Profiler DLL is also responsible for injecting
+the **Loader** (see above) during the application start up.
 
 ![Overview](./images/architecture-overview.png)
 
-### Injecting the OpenTelemetry .NET SDK and instrumentations
+### Bootstrapping
+
+While the initial mechanism for bootstrapping the OpenTelemetry .NET SDK differs
+between .NET and .NET Framework applications, giving that the [host startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md)
+is not available on the .NET Framework,
+in both cases the initialization is done by creating one instance of the
+`OpenTelemetry.AutoInstrumentation.Loader.Startup` class from the Loader assembly.
+When creating this instance the static constructor of the type performs
+the following actions:
+
+- Adds a handler to the
+[`AssemblyResolve` event](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=net-5.0),
+so that it can add any assembly needed by the SDK itself or by any instrumentation.
+- Runs, through reflection, the `Initialization` method from
+  the `OpenTelemetry.AutoInstrumentation.Instrumentation` type
+  from the Managed Profiler assembly.
+  - The `Initialization` code bootstraps the OpenTelemetry .NET SDK,
+    adding configured processors, exporters, and so on,
+    and setting the mechanisms to enable any configured source instrumentations.
+
+#### .NET bootstrapping
+
+.NET applications rely on the [host startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md)
+being configured to use the `StartupHook.Initialize()` method from
+the `OpenTelemetry.AutoInstrumentation.StartupHook` assembly.
+The `Initialize` method will load the `OpenTelemetry.AutoInstrumentation.Loader`
+assembly and will create the `OpenTelemetry.AutoInstrumentation.Loader.Startup` instance.
+
+#### .NET Framework bootstrapping
+
+.NET Framework bootstrapping is performed on the [CorProfiler::JITCompilationStarted](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilercallback-jitcompilationstarted-method)
+callback. When that callback happens the CLR Profiler DLL takes
+the following actions:
+
+- If this the first user module in the current AppDomain,
+the profiler injects the IL calling the Loader `Startup` constructor.
+- If the first method observed by JITCompilationStarted is IIS startup code,
+  the profiler invokes
+  `AppDomain.CurrentDomain.SetData("OpenTelemetry_IISPreInitStart", true)`,
+  so that automatic instrumentations can correctly handle IIS startup scenarios.
```

Comment:
```suggestion
1. If the module is the first user module in the current AppDomain,
the profiler injects the IL to call the Loader `Startup` constructor.
- If the first method observed by `JITCompilationStarted` is IIS startup code,
  the profiler invokes
  `AppDomain.CurrentDomain.SetData("OpenTelemetry_IISPreInitStart", true)`,
  so that automatic instrumentations can correctly handle IIS startup scenarios.
```

(I'm guessing it's a sequence.)